acceptance/lib/facter/acceptance/base_fact_utils.rb

```diff
@@ -0,0 +1,613 @@
+            'os.release.major'         => os_release_major,
+            'os.release.minor'         => os_release_minor,
+            'processors.count'         => /[1-9]/,
+            'processors.physicalcount' => /[1-9]/,
+            'processors.isa'           => /unknown|#{os_hardware}/,
+            'processors.models'        => /"Intel\(R\).*"/,
+            'kernel'                   => 'Linux',
+            'kernelrelease'            => /#{os_kernel}\.\d+/,
+            'kernelversion'            => /#{os_kernel}\.\d+/,
+            'kernelmajversion'         => /#{os_kernel}/
+        }
+        expected_facts
+      end
+
+      # Debian
+      def debian_expected_facts(agent)
+        os_version = agent['platform'].match(/debian-(\d)/)[1]
+        if agent['platform'] =~ /debian-7/
+          codename  = 'wheezy'
+          os_kernel = '3.2'
+        elsif agent['platform'] =~ /debian-8/
+          codename  = 'jessie'
+          os_kernel = '3.16'
+        else
+          fail_test "Unknown Debian version #{agent['platform']}"
+        end
+        if agent['platform'] =~ /amd64/
+          os_arch     = 'amd64'
+          os_hardware = 'x86_64'
+        else
+          os_arch     = 'i386'
+          os_hardware = 'i686'
+        end
+
+        expected_facts = {
+            'os.architecture'          => os_arch,
+            'os.distro.codename'       => codename,
+            'os.distro.description'    => /Debian GNU\/Linux #{os_version}\.\d/,
+            'os.distro.id'             => 'Debian',
+            'os.distro.release.full'   => /#{os_version}\.\d/,
+            'os.distro.release.major'  => os_version,
+            'os.distro.release.minor'  => /\d/,
+            'os.family'                => 'Debian',
+            'os.hardware'              => os_hardware,
+            'os.name'                  => 'Debian',
+            'os.release.full'          => /#{os_version}\.\d/,
+            'os.release.major'         => os_version,
+            'os.release.minor'         => /\d/,
+            'processors.count'         => /[1-9]/,
+            'processors.physicalcount' => /[1-9]/,
+            'processors.isa'           => /unknown|#{os_hardware}/,
+            'processors.models'        => /"Intel\(R\).*"/,
+            'kernel'                   => 'Linux',
+            'kernelrelease'            => os_kernel,
+            'kernelversion'            => os_kernel,
+            'kernelmajversion'         => os_kernel
+        }
+        expected_facts
+      end
+
+      # el (RedHat, Centos)
+      def el_expected_facts(agent)
+        case agent['platform']
+          when /(el|centos)-5/
+            os_version     = '5'
+            kernel_version = '2.6'
+          when /(el|centos)-6/
+            os_version     = '6'
+            kernel_version = '2.6'
+          when /(el|centos)-7/
+            os_version = '7'
+            agent['template'] =~ /oracle/ ? kernel_version = '3.8' : kernel_version = '3.1'
```

Comment:
This is an example of my earlier point. Managing kernel versions for derivatives is going make these tests brittle and error prone. As an example, the validation here for RHEL 7 is incorrect, the kernel version that currently ships with RHEL 7 is '3.10', not '3.1'